hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/performance/ITestS3ADeleteCost.java

```diff
@@ -181,6 +184,8 @@ public void testDeleteFileInDir() throws Throwable {
 
   @Test
   public void testDirMarkersSubdir() throws Throwable {
+    LogCapturer logCapturer =
```

Comment:
i wanted to add test in `TestHttpReferrerAuditHeader` initially but since i thought it would be better to capture headers with an already written tests that perform delete/rename with bulk deletes, and log capturer is the only utility we have to target a specific log name. 
It might look ugly in the sense that we will have to capture a particular line of the logs so if source code changes log lines, then test would fail. but otherwise, it adds stdout as log4j appender, hence logs are always captured, no matter how many tests we run in parallel.

However, i see your point that it's still ugly and not well managed. let me check if there are any other ways to test the same.